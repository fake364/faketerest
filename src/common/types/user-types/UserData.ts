export type UserData = {
  email: string;
  firstName?: string;
  lastName?: string;
};
