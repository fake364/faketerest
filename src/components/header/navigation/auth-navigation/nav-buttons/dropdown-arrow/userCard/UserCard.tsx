import React from 'react';
import ButtonDropdownElement from '../../../../../../../common/components/buttons/buttonDropdown/ButtonDropdownElement';
import { UserData } from '../../../../../../../common/types/user-types/UserData';
import { useSelector } from 'react-redux';
import { RootState } from '../../../../../../../redux/types';

type Props = { onClick: (username: string) => void };

const UserCard: React.FC<Props> = ({ onClick }) => {
  const userData: UserData = useSelector(
    (state: RootState) => state.userData.userData
  );

  const onUserClick = () => onClick(userData.username);

  return (
    <ButtonDropdownElement onClick={onUserClick}>
      <div className="flex gap-[8px]">
        <div className="w-[60px] h-[60px]">
          <div className="w-full h-full rounded-full bg-[whitesmoke] flex justify-center items-center text-[24px]">
            {userData.firstName?.[0].toUpperCase()}
          </div>
        </div>
        <div className="flex flex-col gap-[4px]">
          <div>{`${userData.firstName} ${userData.lastName || ''}`.trim()}</div>
          <div className="font-normal text-[14px]">{userData.email}</div>
        </div>
      </div>
    </ButtonDropdownElement>
  );
};

export default UserCard;
