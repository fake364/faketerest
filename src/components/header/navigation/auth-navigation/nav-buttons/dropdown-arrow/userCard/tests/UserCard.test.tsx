import React from 'react';
import { shallow } from 'enzyme';
import * as redux from 'react-redux';
import { UserData } from '../../../../../../../../common/types/user-types/UserData';
import UserCard from '../UserCard';

jest.mock('react-redux');

describe('UserCard', () => {
  it('should display a bit of user info in as a card', () => {
    jest.spyOn(redux, 'useSelector').mockImplementation((callback) =>
      callback({
        userData: {
          userData: {
            email: 'mail@mail.com',
            lastName: 'Last',
            firstName: 'First',
            username: 'user'
          } as UserData
        }
      })
    );
    const wrapper = shallow(<UserCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
