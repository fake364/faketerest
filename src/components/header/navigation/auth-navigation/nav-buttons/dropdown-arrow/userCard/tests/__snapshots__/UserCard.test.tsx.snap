// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard should display a bit of user info in as a card 1`] = `
<ButtonDropdownElement
  onClick={null}
>
  <div
    className="flex gap-[8px]"
  >
    <div
      className="w-[60px] h-[60px]"
    >
      <div
        className="w-full h-full rounded-full bg-[whitesmoke] flex justify-center items-center text-[24px]"
      >
        F
      </div>
    </div>
    <div
      className="flex flex-col gap-[4px]"
    >
      <div>
        First Last
      </div>
      <div
        className="font-normal text-[14px]"
      >
        mail@mail.com
      </div>
    </div>
  </div>
</ButtonDropdownElement>
`;
